### ISSUE TITLE
Resizing with +/- not working for the non-active window

### ISSUE THREAD

**Issue by dstein64** [vim/vim#5443]
**Description**

Using +/- with `:{winnr}resize` does not work when issued for a non-active window (a window without the cursor).

**To Reproduce**

Detailed steps to reproduce the behavior:
1. Run `vim --clean`
2. Split the window with `:split` and leave the cursor in window 1 (the top window).
3. Type `:2resize +1`. Window 2 (the bottom window) gets larger, as expected.
4. Type `:2resize +1`. Window 2 gets smaller, unexpectedly.

(the same issue occurs with vertical splits and resizing other windows with `:vertical`)

(depending on the initial number of lines when you launch vim, results from steps 3 and 4 above may be swapped)

**Expected behavior**

It's expected that each of these commands will make the second (bottom) window larger. The second command shrinks the second window (presumably the new size for window 2 is based on window 1's size plus one, which is why the first call appears to work and the second call does not work).

The expectation is based on the following documentation in `windows.txt`.

```
:{winnr}res[ize] [+-]N
		Like `:resize` above, but apply the size to window {winnr}
		instead of the current window.
```

**Environment (please complete the following information):**
 - Vim version 8.2
 - OS: macOS 10.15
 - Terminal: Terminal.app

**Comment by lacygoill** [comment]
FWIW, I noticed the same issue when trying to fix a bug in [`vim-dirvish`](https://github.com/justinmk/vim-dirvish/pull/150):

> ... I'm not even sure what :resize does when it receives a prefix count and a relative number as the suffix count; try this:

>     $ vim -Nu NONE +'bo sp|wincmd _'
>     :1resize -5
>     :1resize -5

> Notice how the first :1resize -5 makes the first window occupy the whole terminal minus 5 lines, and how the second :1resize -5 squashes it again. That's not what :resize -5 does in the current window.

As a temporary workaround, [`win_execute()`](https://github.com/vim/vim/releases/tag/v8.1.1418) can be invoked:

    $ vim --clean
    :split
    :call win_execute(win_getid(2), 'resize +1')
    :call win_execute(win_getid(2), 'resize +1')

**Comment by brammool** [comment]
Fixed with patch 8.2.1856
